features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/telemetry/TelemetryWrapper.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.telemetry
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.sampling.RateBasedSampler
+import com.datadog.android.core.sampling.Sampler
+import com.datadog.android.sessionreplay.internal.recorder.telemetry.MethodCalledTelemetry.Companion.METHOD_CALL_OPERATION_NAME
+
+internal class TelemetryWrapper(
```

Comment:
> [!CAUTION]
> This class and is a bit confusing. 
> This class is _technically_ a wrapper, but the name here doesn't give us any insights on what it does. It feels like its main purpose is to create a base metric entity, and IMO doesn't justify creating a class just for that. Besides, using a string as parameter to know what instance to create can lead to issues. If we do use this at all, we should create at least an enum to prevent bad values and get compiler warnings to handle every case).
>  I don't think it's that necessary. Instead I would feel it more relevant to use a factory method in the  `MetricBase`'s companion object.
> E.g.: 

```kotlin
internal interface MetricBase {

    companion object {
        fun startMethodCallMetric(logger: InternalLogger, callerClass: String, samplingRate: Float = 100.0f) : MetricBase? {
        }
    }
}
```